### ISSUE TITLE
CLI reports missing semicolon on yield statements

### ISSUE THREAD

**Issue by pornel** [jshint/jshint#2968]
Given this code:

```js
var x = function*() {
    yield 1;
};
```

and config:

```js
{
    "asi":true,
    "esnext":true,
    "node":true
}
```

When running

```sh
 ./node_modules/.bin/jshint -c conf test.js
```

I'm getting:

> test.js: line 2, col 10, Missing semicolon.

The line with `yield` *does* have a semicolon, so it seems like a parse error in jshint. <del>Oddly, this doesn't happen in jshint on [the website](http://jshint.com)</del>.

```
node v6.2.0
└─┬ jshint@2.9.2
  ├─┬ cli@0.6.6
  │ └─┬ glob@3.2.11
  │   ├── inherits@2.0.1
  │   └─┬ minimatch@0.3.0
  │     ├── lru-cache@2.7.3
  │     └── sigmund@1.0.1
  ├─┬ console-browserify@1.1.0
  │ └── date-now@0.1.4
  ├── exit@0.1.2
  ├─┬ htmlparser2@3.8.3
  │ ├── domelementtype@1.3.0
  │ ├── domhandler@2.3.0
  │ ├─┬ domutils@1.5.1
  │ │ └─┬ dom-serializer@0.1.0
  │ │   ├── domelementtype@1.1.3
  │ │   └── entities@1.1.1
  │ ├── entities@1.0.0
  │ └─┬ readable-stream@1.1.14
  │   ├── core-util-is@1.0.2
  │   ├── isarray@0.0.1
  │   └── string_decoder@0.10.31
  ├── lodash@3.7.0
  ├─┬ minimatch@2.0.10
  │ └─┬ brace-expansion@1.1.5
  │   ├── balanced-match@0.4.1
  │   └── concat-map@0.0.1
  ├── shelljs@0.3.0
  └── strip-json-comments@1.0.4
```

**Comment by nicolo-ribaudo** [comment]
This can be reproduced on the website:
```js
// jshint asi: true, esversion: 6, node: true
var x = function*() {
    yield 1;
};
```

PS. `esnext` is deprecated, you should use `esversion: 6`

**Comment by fullsailor** [comment]
Ran into this today. Strangely enough adding parens prevents the error.

// jshint asi: true, esversion: 6, node: true
var x = function*() {
    yield (1);
};

**Comment by nicolo-ribaudo** [comment]
@jugglinmike Is this fixed by #2978

**Comment by jugglinmike** [comment]
It is indeed, making this issue effectively a duplicate of gh-2923. Expect the fix in the version of JSHint that follows 2.9.2. Coming soon!
